### PR TITLE
Ax/db overhaul

### LINKED ISSUE #92: Update Tricklist and Database

### PR DESCRIPTION
**Pull request by weberax**
this should be the database structure for the beta release ...
closes #92

### CODE REVIEW

File: src/services/db.js [hunk 1]
```diff
@@ -9,136 +10,153 @@ export default class Database {
...
-  };
+  // create or update userTrick
+  saveTrick = (trick) => this.db.userTricks.put(trick);
 
   // delete trick
-  deleteTrick = (id) => {
-    if (id < 1000) console.log("can't delete this trick");
-    return this.db.userTricks.delete(Number(id));
-  };
+  deleteTrick = (id) => this.db.userTricks.put({"id": Number(id), deleted: true});
```
Comment (bastislack) [inline src/services/db.js:98]: This means the user can not delete predefined tricks?

**Review by bastislack — APPROVED** [review]

File: src/predifinedTricksCombos.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+10164,,Triple Back Yoda Roll,Davis Hermes,2021,https://www.instagram.com/p/CUNMWEgAGP4/,,,EXPOSURE,EXPOSURE,11,Three Back Yoda Rolls in one bounce.,
+10165,Holy Grail,Barrel Almighty Flip to Barrel Roll,Davis Hermes,2021,https://youtu.be/UdG4DNI0SNk?t=92,92,97,STAND,STAND,11,Start by doing a Barrel Almighty Flip and go for a second roll by grabbing the line and doing a Barrel Roll.,
+10166,,Darth Vader from Feet,Ian Eisenberg,2021,https://youtu.be/HecUpHXRpDw?t=436,436,452,EXPOSURE,EXPOSURE,11,Start in Expo and roll down until you are under the line. Then do a Darth Vader.,
+10167,,Double Back Almighty Flip,Sebastian Egger,2021,https://youtu.be/HecUpHXRpDw?t=547,547,567,EXPOSURE,EXPOSURE,11,Double Back Almighty Flip,Do the first one laid out and tuck for the second one to get good momentum.; Try to get the hook at your ankle. If the line is near your toes it's hard to pull it around for the second roll.
+10168,,Double Front Almighty Flip,Sebastian Egger,2021,https://youtu.be/HecUpHXRpDw?t=780,780,814,EXPOSURE,EXPOSURE,11,Double Front Almighty Flip.,Stay open for the first one and tuck for the second one to get good momentum.`;
+
```
Comment (bastislack) [inline src/predifinedTricksCombos.js:174]: Should I put the same ids in the google sheet?

**Review by bastislack — APPROVED** [review]

File: src/predifinedTricksCombos.js (hunk 2 above)
Comment (bastislack) [reply src/predifinedTricksCombos.js:174]: Sorry for the two approvals!

File: src/services/db.js (hunk 1 above)
Comment (weberax) [reply src/services/db.js:98]: no, the predefined lists are never modified. All the changes the user makes are noted in the user tricks.
If the user wants to delete a predefined trick, it's just marked as deleted in the user tables

File: src/predifinedTricksCombos.js (hunk 2 above)
Comment (weberax) [reply src/predifinedTricksCombos.js:174]: > Should I put the same ids in the google sheet?

yes that would be awesome

File: src/services/db.js (hunk 1 above)
Comment (bastislack) [reply src/services/db.js:98]: Ok, I see! Cool!

File: src/predifinedTricksCombos.js (hunk 2 above)
Comment (bastislack) [reply src/predifinedTricksCombos.js:174]: > > Should I put the same ids in the google sheet?
> 
> yes that would be awesome

Done!

File: src/services/db.js [hunk 3]
```diff
@@ -9,136 +10,153 @@ export default class Database {
...
-      combos: "++id, name, tricks, minDiff, maxDiff, avgDiff, totalDiff, numberOfTricks, establishedBy, yearEstablished, linkToVideo, comments, stickFrequency"
+    this.db.version(2).stores({
+      // to keep track of all the versions
+      versions: "key,version",
+      // this is the table for the "predefined" tricks, the id's will start from 10000 onwards
+      predefinedTricks: "id,alias,technicalName,establishedBy,yearEstablished,linkToVideo,videoStartTime,videoEndTime,startPos,endPos,difficultyLevel,description,tips,stickFrequency",
+      userTricks: "++id,alias,technicalName,establishedBy,yearEstablished,linkToVideo,videoStartTime,videoEndTime,startPos,endPos,difficultyLevel,description,tips,stickFrequency,deleted",
+      predefinedCombos: "id, name, *tricks, minDiff, maxDiff, avgDiff, totalDiff, numberOfTricks, establishedBy, yearEstablished, linkToVideo, comments, stickFrequency",
+      userCombos: "++id, name, *tricks, minDiff, maxDiff, avgDiff, totalDiff, numberOfTricks, establishedBy, yearEstablished, linkToVideo, comments, stickFrequency, deleted"
```
Comment (bastislack) [inline src/services/db.js:20]: Should we add startPos and endPos on the combos as well?
We could use it e.g. for combining combos in the future.

Comment (bastislack) [reply src/services/db.js:20]: In the combos we can also add videoStartTime and videoEndTime.

Comment (weberax) [reply src/services/db.js:20]: both we can easily add in the future. I would focus now in this pull request more on getting the combos working nicely. see #115

**Comment by weberax** [comment]
most of it is done, just the stats...

**Comment by weberax** [comment]
So, I think this is done (or good enough...)
Everything should work now. The only thing I wanted to do but couldn't figure out how to do nicely was computing the stats automatically for the predefined combos. but this we probably don't need.
@bastislack either you can have another look at the latest commits or just merge it directly, I think I found a good solution

**Review by bastislack — APPROVED** [review]

**Pull request merged**